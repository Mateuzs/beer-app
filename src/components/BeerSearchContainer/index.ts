export { default as BeerSearchContainer } from "./BeerSearchContainer";
