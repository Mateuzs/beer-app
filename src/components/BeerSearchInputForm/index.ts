export { default as BeerSearchInputForm } from "./BeerSearchInputForm";
