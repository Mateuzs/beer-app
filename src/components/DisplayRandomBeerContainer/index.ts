export { default as DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
