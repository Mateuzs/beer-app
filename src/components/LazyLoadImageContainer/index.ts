export { default as LazyLoadImageContainer } from "./LazyLoadImageContainer";
