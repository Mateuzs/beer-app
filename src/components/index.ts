export { BeerSearchContainer } from "./BeerSearchContainer";
export { DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
export { NavBar } from "./NavBar";
export { SearchButton } from "./SearchButton";
export { LazyLoadImageContainer } from "./LazyLoadImageContainer";
export { ErrorMessage } from "./ErrorMessage";
export { BeerSearchInputForm } from "./BeerSearchInputForm";
export { BeerList } from "./BeerList";
