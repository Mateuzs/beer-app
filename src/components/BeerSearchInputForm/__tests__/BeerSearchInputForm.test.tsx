// external
import React from "react";
import { shallow, mount } from "enzyme";
// components
import { BeerSearchInputForm } from "../../../components";

describe("BeerSearchInputForm component", () => {
  test("should match the snapshot and styles", () => {
    const mockedData = {
      inputValue: "INPUT_VALUE",
      isValidInputValue: true,
      inputPlaceholder: "INPUT_PLACEHOLDER",
      inputDescription: "INPUT_DESCRIPITON",
      onInputChangeCallback: jest.fn(),
      onInputRadioChangeCallback: jest.fn(),
    };
    const wrapper = shallow(<BeerSearchInputForm {...mockedData} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should render expected components", () => {
    const mockedData = {
      inputValue: "INPUT_VALUE",
      isValidInputValue: true,
      inputPlaceholder: "INPUT_PLACEHOLDER",
      inputDescription: "INPUT_DESCRIPITON",
      onInputChangeCallback: jest.fn(),
      onInputRadioChangeCallback: jest.fn(),
    };
    const component = mount(<BeerSearchInputForm {...mockedData} />);

    expect(component.find('[data-test-id="beer-search-input"]').length).toEqual(1);
    expect(component.find('[data-test-id="beer-search-description"]').length).toEqual(1);
    expect(component.find('[data-test-id="beer-search-input-radio-name"]').length).toEqual(1);
    expect(component.find('[data-test-id="beer-search-input-radio-brewed-before"]').length).toEqual(
      1
    );
  });
});
