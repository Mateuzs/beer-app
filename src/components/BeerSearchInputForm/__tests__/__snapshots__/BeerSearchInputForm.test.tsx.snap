// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerSearchInputForm component should match the snapshot and styles 1`] = `
<div
  className="beer-search-input-container"
>
  <div
    className="beer-search-input-component"
  >
    <input
      className="beer-search-input"
      data-test-id="beer-search-input"
      onChange={[Function]}
      placeholder="INPUT_PLACEHOLDER"
      value="INPUT_VALUE"
    />
    <p
      className="beer-search-description-correct"
      data-test-id="beer-search-description"
    >
      INPUT_DESCRIPITON
    </p>
  </div>
  <div
    className="bear-search-radio-input-container"
  >
    <input
      className="beer-search-radio-input"
      data-test-id="beer-search-input-radio-name"
      defaultChecked={true}
      id="name"
      name="beer-settings"
      onChange={[Function]}
      type="radio"
      value="name"
    />
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="beer-search-radio-input"
      data-test-id="beer-search-input-radio-brewed-before"
      id="brewedBeforeDate"
      name="beer-settings"
      onChange={[Function]}
      type="radio"
      value="brewedBeforeDate"
    />
    <label
      htmlFor="brewedBeforeDate"
    >
      Brewed Before
    </label>
  </div>
</div>
`;
